### PR TITLE
docs: Added architcture docs

### PR DESCRIPTION
**Pull request by christian-kreuzberger-dtx**
Signed-off-by: Christian Kreuzberger <christian.kreuzberger@dynatrace.com>

### CODE REVIEW

**Review by Raffy23 — DISMISSED** [review]
LGTM

File: docs/ARCHITECTURE.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Architecture
+
+Keptn-Gitea-Provisioner-Service is a microservice with a simple job: When a Keptn project without an upstream is created, it creates a git repo and a token on a (remote) Gitea installation, and return the Git Remote URL, username and token, e.g.:
+
+```
+POST /repository
+  project=foobar
```
Comment (Raffy23) [inline docs/ARCHITECTURE.md]: The request that the provisioner has to process is structured a little bit differently: https://keptn.sh/api/#/repository/addInstance

Comment (christian-kreuzberger-dtx) [reply docs/ARCHITECTURE.md]: you are right, this was a little too much simplificaiton on my side. fixed it

**Review by Raffy23 — APPROVED** [review]
LGTM

**Pull request merged**